src/main/java/com/github/onsdigital/babbage/api/endpoint/content/Resource.java

```diff
@@ -23,13 +36,72 @@
     @GET
     public void get(@Context HttpServletRequest request, @Context HttpServletResponse response) throws IOException {
         try {
-            ContentResponse contentResponse = ContentClient.getInstance().getResource(request.getParameter("uri"));
+            String uri = request.getParameter("uri");
+            String width = request.getParameter("width");
+
+            ContentResponse contentResponse = null;
+
+            try {
+                contentResponse = ContentClient.getInstance().getResource(uri);
+            } catch (ContentReadException e) {
+                System.out.println("Get resource returned error response, status: " + e.getStatusCode());
+                ErrorHandler.renderErrorPage(404, response);
+                return;
+            }
+
             String contentDispositionHeader = "inline; ";
             contentDispositionHeader += contentResponse.getName() == null ? "" : "filename=\"" + contentResponse.getName() + "\"";
             response.setHeader("Content-Disposition", contentDispositionHeader);
-            new BabbageContentBasedBinaryResponse(contentResponse, contentResponse.getDataStream(), contentResponse.getMimeType()).apply(request, response);
+
+            try ( InputStream contentResponseBody = contentResponse.getDataStream() ) {
+
+                BufferedImage image = null;
+
+                if (width != null) {
+
+                    try {
+                        Integer w = Integer.parseInt(width, 10);
+
+                        if (uri.toLowerCase().endsWith(".png") || uri.toLowerCase().endsWith(".jpg") || uri.toLowerCase().endsWith(".jpeg")) {
+                            image = ImageIO.read(contentResponseBody);
+                            Integer clampedWidth = Math.max(1, Math.min(image.getWidth(), w));
```

Comment:
Consider using primitive types: `int` instead of `Integer`, and `double` instead of `Double`. Some more detail here: http://jtechies.blogspot.co.uk/2012/07/item-49-prefer-primitive-types-to-boxed.html